### PR TITLE
Handle different data types (int, obj, array, etc) when deploying tem…

### PR DESCRIPTION
**Pull request by vivsriaus**
…plates. Use json.parse for getting template param values

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@vivsriaus__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (visriniv). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by yugangw-msft** [comment]
please fix jlint error by using single quots

File: lib/commands/arm/group/groupUtils._js [hunk 1]
```diff
@@ -371,6 +371,17 @@ function getTemplateParams(cli, subscription, options, _)
     paramKeys[param] = templateContent.parameters[param].defaultValue;
     if (!templateContent.parameters[param].defaultValue) {
       paramKeys[param] = cli.interaction.promptIfNotGiven(param + ': ', paramKeys[param], _);
+      switch (templateContent.parameters[param].type.toLowerCase()) {
+        case "int":
+          paramKeys[param] = utils.parseInt(paramKeys[param]);
+          break;
+
```
Comment (amarzavery) [inline lib/commands/arm/group/groupUtils._js]: In the template mentioned in the issue, the type of instanceCount is string and not int.
So how will the above code solve the problem?
```
error: Deployment template validation failed: 
'The provided value for the template parameter
 'instanceCount' at line '8' and column '28' is not valid.'.
```
```
"parameters": {
"vmSSName": {
"type": "string"
},
"instanceCount": {
"type": "string",
"metadata": {
"description": "Number of VM instances"
}
},
```

**Comment by amarzavery** [comment]
It will be nice, if you can add a unit test that verifies the issue is fixed.

**Comment by amarzavery** [comment]
@vivsriaus - This seems to be a related issue. Can you fix this as well https://github.com/Azure/azure-xplat-cli/issues/1831 ?

File: lib/commands/arm/group/groupUtils._js (hunk 1 above)
Comment (vivsriaus) [inline lib/commands/arm/group/groupUtils._js]: The instanceCount type should be an "int". People were using "string" to get around this issue. So, now, if you edit the template and specify instanceCount as "int", the group deployment command should work.

**Comment by vivsriaus** [comment]
@amarzavery I've modified an existing test to include "int" in template file and have captured new records. With regards to improving the error message, we'll take that as a separate PR.

**Pull request merged**